core/src/entry_point/mod.rs

```diff
@@ -84,7 +84,7 @@ libtock_codegen::make_read_env_var!("APP_HEAP_SIZE");
 /// into the rustc-generated main(). This cannot use mutable global variables or
 /// global references to globals until it is done setting up the data segment.
 #[no_mangle]
-unsafe extern "C" fn rust_start(app_start: usize, stacktop: usize, _app_heap_break: usize) -> ! {
+unsafe extern "C" fn rust_start(app_start: usize, stacktop: usize, app_heap_start: usize) -> ! {
```

Comment:
From looking at the ARM assembly, this looks good to me. I haven't checked against the RISC-V assembly because I'm not sufficiently familiar with RISC-V (I already misread the ARM assembly once).

Reply:
This looks okay to me from the RISC-V side. app_heap_start is already calculated in assembly so we're keeping the same behavior.